src/webextension/survey/components/app.jsx

```diff
@@ -46,7 +46,16 @@ class App extends Component {
   }
 
   handleSubmit(values) {
-    sendMessage('submission', values);
+    Promise
+      .all([
+        browser.tabs.query({ active: true, currentWindow: true }),
+        browser.runtime.getBackgroundPage()
+      ])
+      .then(([ tabs, backgroundWindow ]) => {
+        const requests = backgroundWindow.app.httpObserver.get(tabs[0].id);
```

Comment:
I assume that `tabs[0]` will always be defined and not have potential issues with accessing `.id` of a falsey?
I'm guessing so, since `tabs` array comes via `browser.tabs.query()` which probably wouldn't return an empty array.